src/pybind/mgr/diskprediction_local/tests/test_predictors.py

```diff
@@ -0,0 +1,112 @@
+                            int(attr.get('raw', {}).get('string', '0'))
+                    else:
+                        if str(attr.get('raw', {}).get('string', '0')).split(' ')[0].isdigit():
+                            dev_smart['smart_%s_raw' % attr.get('id')] = \
+                                int(attr.get('raw', {}).get('string',
+                                                            '0').split(' ')[0])
+                        else:
+                            dev_smart['smart_%s_raw' % attr.get('id')] = \
+                                attr.get('raw', {}).get('value', 0)
+                # get normalized smart values
+                if attr.get('value') is not None:
+                    dev_smart['smart_%s_normalized' % attr.get('id')] = \
+                                attr.get('value')
+            # add power on hours manually if not available in smart attributes
+            if s_val.get('power_on_time', {}).get('hours') is not None:
+                dev_smart['smart_9_raw'] = int(s_val['power_on_time']['hours'])
+            # add device capacity
+            if s_val.get('user_capacity') is not None:
+                if s_val.get('user_capacity').get('bytes') is not None:
+                    dev_smart['user_capacity'] = s_val.get('user_capacity').get('bytes')
+                else:
+                    self.log.debug('user_capacity not found in smart attributes list')
+            # add device model
+            if s_val.get('model_name') is not None:
+                dev_smart['model_name'] = s_val.get('model_name')
+            # add vendor
+            if s_val.get('vendor') is not None:
+                dev_smart['vendor'] = s_val.get('vendor')
+            # if smart data was found, then add that to list
+            if dev_smart:
+                self.predict_datas.append(dev_smart)
+            if len(self.predict_datas) >= 12:
+                break
+
+
+    def test_setup(self):
+        """Tests if setup was done correctly 
+        """
+        # ensure test pt is a list of dicts, as it will be for the module
+        self.assertIsInstance(self.predict_datas, list)
+        for day_data in self.predict_datas:
+            self.assertIsInstance(day_data, dict)
+
+        # input to module will have <=12 days and >=6 days of data
+        # ensure that test pt reflects that
+        self.assertLessEqual(len(self.predict_datas), 12)
+        self.assertGreaterEqual(len(self.predict_datas), 6)
+
+
+    def test_red_hat_predictor(self):
+        """Tests if Red Hat created model works correctly
+        """
+        # check if red hat models initialize properly
+        rh_predictor = RHDiskFailurePredictor()
+        ret = rh_predictor.initialize('models/redhat')
+        self.assertEqual(ret, None)
+
+        # predict and check if value is a valid output
+        pred = rh_predictor.predict(self.predict_datas)
+        self.assertIn(pred, self.valid_outputs)
```

Comment:
We should use the assert to verify the specific output we're expecting the model to return given a specific input (instead of any possible outputs the model has in general).

Same in `def test_prophetstor_predictor(self)`.

Reply:
Makes sense, I'll update it :smile:

Reply:
Thanks for being patient, I've updated tests to check for specific output. And also added a check to ensure that models don't crash in case of an invalid json input